### PR TITLE
chore: update libp2p dependencies

### LINKED ISSUE #1714: Use seed nodes as trusted peeers while initializing go-header
TL;DR: we should use seed nodes as trusted peers while initializing go-header. This is in-line with usage in celestia-node, where bootstrappers are used as trusted peers.

----

While working on https://github.com/rollkit/rollkit/pull/1684 I discovered some issues related to peer handling in `go-header`. Those issues are on rollkit side - `go-header` is not used as it supposed to be.

They are visible only on latest releases of `go-libp2p` and `go-header` because of some slight changes related to peer discovery, timing, etc.

Currently full nodes try to discover peers using DHT, and then use them as trusted peers while initializing go-header.
If peers are not discovered properly (and it seems a bit more time is needed with latest libp2p versions) before go-header initialization, `go-header` `Syncer` is not started at all.

To fix this, we should feed our seed nodes as trusted peers for `go-header`.

### LINKED ISSUE #1715: `TestPendingBlocks` in `node` is incorrect
`TestPendingBlocks` in `node` package is not working correctly - it was shown while working on https://github.com/rollkit/rollkit/pull/1684.

The first node run is generating all the blocks. When new node is created for tests, it setups new validator (aggregator) keys, so it's not able to validate old blocks that might be pending (as the signature was generated using old key).

This is strictly issue with the test. To fix it, we need to ensure that same key is used for both aggregator node instances.

### PR DESCRIPTION
**Pull request by tzdybal**
## Overview

Update all the libp2p dependencies to latest releases.

Fixes a lot of issues that appeared at the same time.
Resolves #1714.
Resolves #1715.

## Summary by CodeRabbit

- **New Features**
  - Improved configuration options for node synchronization by adding a `skipPeers` parameter to various node creation functions.

- **Refactor**
  - Updated IP addresses in test setups for better testing consistency.
  - Added conditional checks for appending seed nodes based on configuration parameters.

- **Dependency Updates**
  - Updated multiple dependencies to their latest versions, enhancing security and performance.

- **Tests**
  - Enhanced various test functions to support new configuration parameters, improving flexibility in testing node synchronization and configuration scenarios.

### CODE REVIEW

**Review by yarikbratashchuk — DISMISSED** [review]
Yeah, long overdue :)

**Review by tuxcanfly — DISMISSED** [review]
OK.

**Comment by MSevey** [comment]
@coderabbitai resolve

**Review by MSevey — DISMISSED** [review]

**Review by Manav-Aggarwal — CHANGES_REQUESTED** [review]

File: node/full_node_integration_test.go [hunk 1]
```diff
@@ -350,6 +350,8 @@ func TestSingleAggregatorTwoFullNodesBlockSyncSpeed(t *testing.T) {
 
 	require.NoError(verifyNodesSynced(node1, node2, Store))
 	require.NoError(verifyNodesSynced(node1, node3, Store))
+
+	fmt.Println("tzdybal: elo")
```
[the remark this answers is not in the document]
Comment (tzdybal) [reply node/full_node_integration_test.go:354]: OMG :facepalm:

**Review by Manav-Aggarwal — APPROVED** [review]
LGTM

**Review by MSevey — APPROVED** [review]

**Review by yarikbratashchuk — APPROVED** [review]
looks alright, but TestTxGossipingAndAggregation is failing

**Pull request merged**